include/sys/signal.h

```diff
@@ -81,14 +81,18 @@ typedef struct ctx ctx_t;
  *
  * \sa sig_post
  * \note Must be called with p::p_lock held. Returns with p::p_lock held.
+ *
+ * \returns error code if can't signal process
  */
-void sig_kill(proc_t *p, signo_t sig);
+int sig_kill(proc_t *p, signo_t sig);
```

Comment:
I don't think we want to add permission checking to `sig_kill()`: its responsibility should be to send a signal to a process. We can create another procedure that checks permissions before calling `sig_kill()` or simply check permissions manually before calling it. This is not unreasonable, since we need permission checks only when sending a signal on behalf of a user, i.e. in `sys_kill()`. NetBSD does manual permission checks as needed before calling [`kpsignal2()`](http://bxr.su/NetBSD/sys/kern/kern_sig.c#kpsignal2): see [`kill1()`](http://bxr.su/NetBSD/sys/kern/sys_sig.c#kill1).